app/src/main/java/com/example/wrk/models/WorkoutPerformed.java

```diff
@@ -25,4 +29,38 @@ public void setUser(ParseUser parseUser) {
     public WorkoutTemplate getWorkout() { return (WorkoutTemplate) getParseObject(KEY_WORKOUT); }
     public void setWorkout(WorkoutTemplate workout) { put(KEY_WORKOUT, workout); }
 
+    private static final int SECOND_MILLIS = 1000;
+    private static final int MINUTE_MILLIS = 60 * SECOND_MILLIS;
+    private static final int HOUR_MILLIS = 60 * MINUTE_MILLIS;
+    private static final int DAY_MILLIS = 24 * HOUR_MILLIS;
+
+    public String getRelativeTimeAgo(String rawJsonDate) {
+        String timeFormat = "EEE MMM dd HH:mm:ss ZZZZZ yyyy";
+        SimpleDateFormat sf = new SimpleDateFormat(timeFormat, Locale.ENGLISH);
+        sf.setLenient(true);
+        try {
+            long time = sf.parse(rawJsonDate).getTime();
+            long now = System.currentTimeMillis();
+
+            final long diff = now - time;
+            if (diff < MINUTE_MILLIS) {
+                return "just now";
+            } else if (diff < 2 * MINUTE_MILLIS) {
+                return "a minute ago";
+            } else if (diff < 50 * MINUTE_MILLIS) {
+                return diff / MINUTE_MILLIS + "m";
+            } else if (diff < 90 * MINUTE_MILLIS) {
+                return "an hour ago";
+            } else if (diff < 24 * HOUR_MILLIS) {
+                return diff / HOUR_MILLIS + "h";
+            } else if (diff < 48 * HOUR_MILLIS) {
+                return "yesterday";
+            } else {
+                return diff / DAY_MILLIS + "d";
```

Comment:
I find it might be better to use existing APIs for this kind of processing, unless you want it super custom for your particular usecase. For example, I think you might be able to use Android `DateUtils.getRelativeTimeSpanString`

Reply:
> I find it might be better to use existing APIs for this kind of processing, unless you want it super custom for your particular usecase. For example, I think you might be able to use Android `DateUtils.getRelativeTimeSpanString`

I can look into this. I had just used this being that I had used this on one of my previous projects to get the timestamp.